src/commands/chat/ui/AiResponseLoader.tsx

```diff
@@ -1,14 +1,27 @@
 import React from 'react';
-import { Box, Text } from 'ink';
-import { Spinner } from '@inkjs/ui';
+import { Text } from 'ink';
+import { TextSpinner } from '../../../components/TextSpinner.js';
 import { colors } from '../../../theme/colors.js';
 import { texts } from '../texts.js';
 
-export function AiResponseLoader() {
+interface AiResponseLoaderProps {
+  text?: string;
+}
+
+export function AiResponseLoader({ text }: AiResponseLoaderProps) {
+  if (text) {
+    return (
+      <Text color={colors.assistant}>
+        <Text color={colors.assistant}>{texts.assistantLabel}</Text>
+        <Text>{text} </Text> <TextSpinner type="sand" />
+      </Text>
+    );
+  }
+
   return (
-    <Box flexDirection="row">
+    <Text color={colors.assistant}>
       <Text color={colors.assistant}>{texts.assistantLabel}</Text>
-      <Spinner type="sand" label={texts.responseLoading} />
-    </Box>
+      <TextSpinner type="sand" /> {texts.responseLoading}
+    </Text>
```

Comment:
This remounts the root Text when `text` changes. Maybe this would work?
```jsx
return (
      <Text color={colors.assistant}>
        <Text color={colors.assistant}>{texts.assistantLabel}</Text>
        {text
          ? <Text>{text} </Text> <TextSpinner type="sand" />
          : <TextSpinner type="sand" /> {texts.responseLoading}
      </Text>
    );
```

Reply:
btw, is there Suspense for ink? would be even better

Reply:
You're right, root element returned would be the same (Text). Remount would happen if both returns had different elements and I got that wrong here